deployment/address_book.go

```diff
@@ -82,14 +85,16 @@ type AddressBook interface {
 	AddressesForChain(chain uint64) (map[string]TypeAndVersion, error)
 	// Allows for merging address books (e.g. new deployments with existing ones)
 	Merge(other AddressBook) error
+	Remove(ab AddressBook) error
 }
 
 type AddressBookMap struct {
 	AddressesByChain map[uint64]map[string]TypeAndVersion
```

Comment:
If we're wrapping locking code around accesses to this, should we allow this to be public? There's nothing stopping someone from just directly accessing this internal storage and bypassing the locking mechanism, is there?

Reply:
Missed this, thank you. I'll fix